File: internal/astgrep/testdata/fixtures/php/violation.php
<?php
// PHP violation fixture — demonstrates patterns matched by php rules.
// This file should produce >= 1 finding when scanned with the php rule set.

// Matches php-unused-var: function returning null without type annotation
function getBadResult(): mixed {
    return null;
}

// Matches php-null-deref: json_decode with null argument
$data = json_decode(null);

// Matches php-todo-marker: stub using die()
function notImplemented(): void {
    die("TODO");
}

File: internal/astgrep/testdata/fixtures/kotlin/violation.kt
// Kotlin violation fixture — demonstrates patterns matched by kotlin rules.
// This file should produce >= 1 finding when scanned with the kotlin rule set.

// Matches kotlin-unused-var: val initialized to null
val x = null

// Matches kotlin-null-deref: non-null assertion operator !!
val value: String? = null
val len = value!!.length

// Matches kotlin-todo-marker: TODO() stub
fun notImplemented() = TODO()
